### DISCUSSION TITLE
One way I use this software to aid in my instruction

### DISCUSSION THREAD

**Discussion by gutow** [gutow/Algebra_with_Sympy#6] [Show and tell]
As a professor of chemistry one of the things I do a lot is produce examples of how to solve physical science problems that involve mathematics. These are either solutions to assigned problems or used for handouts and exercises. This tool facilitates rapidly producing typeset examples, while also allowing me to be explicit about what manipulation was done at each step. I primarily do this using Jupyter notebooks, because I can also intersperse textual explanations (markdown cells), images and graphs. I generally paste screen snapshots into documents to produce .pdfs for handouts. However, I am shifting towards generating the .pdfs directly from the notebooks. Below is an image of a very simple example of a solution to a problem using the ideal gas law. I have kept the example simple involving nothing more than addition, subtraction, multiplication and division. I regularly use this tool to do examples involving calculus. The full functionality of [sympy](https://sympy.org) is available.
# Example
Assume the conditions stay within the range where the gas sample behaves ideally (it follows the ideal gas law: $pV=nRT$, where p = pressure, V = volume, n = moles, R = gas constant, and T = temperature in Kelvin). Initially the gas sample is at 295 K and 760 Torr. If the volume of the gas sample triples while the pressure drops to 700 Torr, what is the final temperature of the gas?
## Showing the code to make the steps completely clear:
![image](https://user-images.githubusercontent.com/1231317/174461822-d4154553-4fed-443f-a4d8-61624c7e0641.png)
![image](https://user-images.githubusercontent.com/1231317/174461892-80a1711f-41ae-47cb-9996-6a38c8d07279.png)

## Code hidden to look more like a normal answer key:
![image](https://user-images.githubusercontent.com/1231317/174461994-bd7690da-7907-4c18-8d16-f91c61dc7d7e.png)

## Comment
For more advanced classes, I would not explain in words what algebra is done at each step. However, showing the code in the notebook automatically provides the students with that information.

**Comment by PascalGitz** [comment]
Hey

First of all thank you !

I just stumbled across algebra_with_sympy due to my post on [Stackoverflow](https://stackoverflow.com/questions/76967456/proper-workflow-for-storing-symbols-and-expressions/76969598#76969598), and it fits my workflow nearly perfect as I'm having similar tasks to do in my structural engineering profession. Especially the Eqn- class is amazing, which speeds up the calculation process drastically compared to regular sympy.

Nontheless I usually use [Quarto ](https://quarto.org/)to convert my Notebooks to readable PDFs wich enabled me to crossrefence between images and equations, set imagecaptions and so on. With sympy i was able to use `sp.init_printing(use_latex='mathjax', latex_mode='equation')` which then produced numbered equations in the latex export. Importing your package leads to overwrite this setting. Is it possible that you add the `latex_mode='equation'`.
